app/models/performance_statistics.rb

```diff
@@ -15,21 +15,72 @@ class PerformanceStatistics
           c.id
   )
   SELECT
-      COUNT(*) AS total_non_dfe_sign_ups,
+      COUNT(*) AS total_candidate_count,
       SUM(CASE WHEN candidate_forms = 0 THEN 1 ELSE 0 END) AS candidates_signed_up_but_not_signed_in,
       SUM(CASE WHEN candidate_forms > 0 AND candidate_started_form_count = 0 AND candidate_submitted_form_count = 0 THEN 1 ELSE 0 END) AS candidates_signed_in_but_not_entered_data,
       SUM(CASE WHEN candidate_started_form_count > 0 AND candidate_submitted_form_count = 0 THEN 1 ELSE 0 END) AS candidates_with_unsubmitted_forms,
       SUM(CASE WHEN candidate_submitted_form_count > 0 THEN 1 ELSE 0 END) AS candidates_with_submitted_forms
   FROM
       raw_data".freeze
 
+  APPLICATION_FORM_STATUS_COUNTS_QUERY = "
+  WITH raw_data AS (
+      SELECT
+          f.id,
+          COUNT(f.id) FILTER (WHERE f.id IS NOT NULL) application_forms,
+          COUNT(ch.id) FILTER (WHERE f.id IS NOT NULL) application_choices,
+          CASE
+            WHEN ARRAY_AGG(DISTINCT ch.status) = '{NULL}' THEN ARRAY['0', 'unsubmitted']
+            WHEN ARRAY_AGG(DISTINCT ch.status) = '{unsubmitted}' THEN ARRAY['0', 'unsubmitted']
+            WHEN 'awaiting_references' = ANY(ARRAY_AGG(ch.status)) THEN ARRAY['1', 'awaiting_references']
+            WHEN 'application_complete' = ANY(ARRAY_AGG(ch.status)) THEN ARRAY['2', 'waiting_to_be_sent']
```

Comment:
a couple of humble questions (my sql is very much less good than yours):

- why do we need to check for null statuses? I think `null` would break the state machine?
- if I understand `ARRAY_AGG(ch.status)` right, this creates an array (denoted `{item item}`) of the status of the attached choices. In the case of `unsubmitted`, `awaiting_references` and `application_complete` the choices should move forward in lockstep (see also `TestApplications.create_application`), so is the `ANY` necessary in these cases?

Reply:
> why do we need to check for null statuses? I think null would break the state machine?

I believe this is to handle the case where the application form hasn't been created at all. If I remove it, the match against `ProcessState` doesn't work for all applications.

> if I understand ARRAY_AGG(ch.status) right, this creates an array (denoted {item item}) of the status of the attached choices. In the case of unsubmitted, awaiting_references and application_complete the choices should move forward in lockstep (see also TestApplications.create_application), so is the ANY necessary in these cases?

I did a direct port of the Ruby logic in `ProcessState` and would be a bit uncomfortable to have them diverge. I think once we prove that this approach is useful, we probably want to do something like materialising the application form state onto the record itself and updating it each time the application choices change.

Reply:
No further questions. Thank you!